websockets/client.py

```diff
@@ -35,9 +35,8 @@ def write_http_request(self, path, headers):
 
         """
         self.path = path
-        self.request_headers = email.message.Message()
-        for name, value in headers:
-            self.request_headers[name] = value
+        self.request_headers = http.client.HTTPMessage()
+        self.request_headers._headers = headers     # HACK
```

Comment:
Yes, I ended up using `http.client.HTTPMessage` a bit reluctantly and I'm not very happy with that hack.

The constraints here are:

- not losing any information: I believe the the `raw_*_headers` lists of `(name, value)` pairs are a good way to achieve that; I'm happy with them (even though it's unlikely that they will be used widely)
- providing a convenient API for accessing headers: we're immediately getting into the problem of multi-value dicts where 99,99% of accesses expect a single value; each library tends to invent its own API; I decided to use the closest matching data structure in the standard library to avoid adding one more variant to the Python ecosystem
- minimizing performance overhead; the HACK is the only way I found to instantiate a `HTTPMessage` without running a lot of code; it don't expect the implementation to change, but the risk exists in theory; I believe tests cover this
- not reinventing the wheel: not only does `http.client.HTTPMessage` feel semantically correct for representing HTTP headers, but it's backwards compatible because it inherits `email.message.Message` (the less semantically correct type previously used there); see also https://docs.python.org/3/library/http.client.html#http.client.HTTPResponse.msg

I considered using properties and rejected that idea because it would make the code more verbose, slightly less efficient, and make it less obvious where these values are set.